lightning/src/ln/onion_utils.rs

```diff
@@ -183,10 +183,10 @@ fn shift_arr_right(arr: &mut [u8; ONION_DATA_LEN], amt: usize) {
 pub(super) fn route_size_insane(payloads: &Vec<msgs::OnionHopData>) -> bool {
 	let mut len = 0;
 	for payload in payloads.iter() {
-		let mut payload_len = LengthCalculatingWriter(0);
+		let mut payload_len = Writer::new();
```

Comment:
I thought only the legacy onion hop is fixed length?

Reply:
Correct, I was referring there to the idea of adding a function to `Writeable` (or maybe adding a new trait) which allows a caller to ask for the exact size of the object once serialized. We could implement it for a few network messages and use it here.

Reply:
Added to `Writeable` and implemented for the `OnionHopData`.

Anywhere else we could implement it where it would make a significant difference?  I would guess many network messages don't happen often enough to matter.